app/src/main/java/com/example/interesting_facts_about_numbers/ui/presentation/number_fact_screen/NumberFactViewModel.kt

```diff
@@ -0,0 +1,56 @@
+package com.example.interesting_facts_about_numbers.ui.presentation.number_fact_screen
+
+import androidx.lifecycle.SavedStateHandle
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import androidx.navigation.NavHostController
+import com.example.interesting_facts_about_numbers.functionality.GetNumberFactByNumberUseCase
+import com.example.interesting_facts_about_numbers.functionality.GetRandomNumberFactUseCase
+import dagger.hilt.android.lifecycle.HiltViewModel
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.StateFlow
+import kotlinx.coroutines.launch
+import javax.inject.Inject
+
+@HiltViewModel
+class NumberFactViewModel @Inject constructor(
+    savedStateHandle: SavedStateHandle,
+    private val getRandomNumberFactUseCase: GetRandomNumberFactUseCase,
+    private val getNumberFactByNumberUseCase: GetNumberFactByNumberUseCase,
+) : ViewModel() {
+    private val mutState: MutableStateFlow<NumberFactUiState> =
+        MutableStateFlow(
+            NumberFactUiState.Initial.Loading
+        )
+
+    val state: StateFlow<NumberFactUiState> = mutState
+
+    init {
+        val pasArg: String = checkNotNull(savedStateHandle["number"])
```

Comment:
You created nullable param and then check it here if it's not null

Reply:
it was fixed